### ISSUE TITLE
run_evaluation() does not work.

### ISSUE THREAD

**Issue by xingzhaolee** [PyTorchLightning/pytorch-lightning#763]
## 🐛 Bug

`run_evaluation()` does not work. Suspected that model is not loaded into the trainer at any point.

### To Reproduce

1. Run ImageNet example with `--evaluate` argument. `python imagenet_example.py --evaluate`

### Expected behavior

Model is supposed to load from the checkpoint directory and evaluation on validation is carried out.

**Comment by sneiman** [comment]
does this fail in every case or only with ddp back end?

**Comment by xingzhaolee** [comment]
@sneiman the ImageNet example uses the wrong method. `run_evaluation` only works when at least 1 epoch of training is used. `trainer.test()` should be used instead. Also I'm working on a pull request to separate `.validate()` and `.test()` since in cases like ImageNet, there's no test set or in the event that user will want to evaluate on validation set only.

**Comment by williamFalcon** [comment]
@xingzhaolee awesome. Excited for the PR. Let's get it into this release.

**Comment by williamFalcon** [comment]
@xingzhaolee I now realize ```run_evaluation()``` is a private method... it's not intended to be used directly by the user. 
@Borda

**Comment by Borda** [comment]
well, then it shall be at least `_run_evaluation()` :]

**Comment by xingzhaolee** [comment]
misunderstanding was caused by imagenet example using `run_evaluation()`

**Comment by daniilhayrapetyan** [comment]
Are there any updates on this? The separation of `.validate` and `.test` seems very useful.

**Comment by williamFalcon** [comment]
can you give an example of what you would like to do?
